### PR TITLE
Add required library to project configuration

### PR DESCRIPTION
**Pull request by alecharp**
Update minimum core version and bom

Since https://github.com/jenkinsci/jenkins/pull/5047 was merged in Jenkins Core, the library `libzfs` was removed. This library is required in this plugin but was not properly described. 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -209,6 +209,7 @@ THE SOFTWARE.
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>script-security</artifactId>
+            <version>1.77</version>
```
Comment (MRamonLeon) [inline pom.xml:212]: can't we use the one from the bom?

File: pom.xml [hunk 2]
```diff
@@ -287,11 +286,16 @@ THE SOFTWARE.
...
-                <artifactId>bom-2.222.x</artifactId>
-                <version>29</version>
+                <artifactId>bom-2.277.x</artifactId>
+                <version>807.v6d348e44c987</version>
                 <scope>import</scope>
                 <type>pom</type>
             </dependency>
+            <dependency>
+                <groupId>joda-time</groupId>
+                <artifactId>joda-time</artifactId>
```
Comment (MRamonLeon) [inline pom.xml:296]: why is it needed now and not before?

**Review by MRamonLeon — COMMENTED** [review]
Some doubts

File: pom.xml (hunk 1 above)
Comment (alecharp) [reply pom.xml:212]: once https://github.com/jenkinsci/bom/commit/0de33628090ece61378f6e8d97711b93af9b997a is in a release of the BOM, yes.

File: pom.xml (hunk 2 above)
Comment (alecharp) [reply pom.xml:296]: Because of the version used in two different dependencies which are using different versions.

**Review by MRamonLeon — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (alecharp) [reply pom.xml:212]: I can add a todo to remove the version once BOM is released

**Review by res0nance — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (MRamonLeon) [reply pom.xml:212]: it would be great @alecharp , thanks!

Comment (alecharp) [reply pom.xml:213]: done in https://github.com/jenkinsci/ec2-plugin/pull/615/commits/06887408f6d0d7cc77c76fd6c74565b19df940d3

**Pull request merged**